src/main.rs

```diff
@@ -62,6 +63,7 @@ fn start_speaker(mic_cfg: rodio::SupportedStreamConfig, cfg: &Config) {
 
     let local_addr = format!("0.0.0.0:{}", cfg.port);
     let recv_socket = net::UdpSocket::bind(local_addr).expect("Failed to bind recv socket");
+    recv_socket.set_read_timeout(dur);
```

Comment:
idk why I did this?

Reply:
I think I would rather have this timeout handled in state machine anyway

Reply:
should just set non blocking i think. I did that on my ring pr actually https://github.com/tanner0101/piphone/pull/2/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fcR74

Reply:
I am not familiar with this right now - is it blocking as is?

Reply:
Without the set read timeout?

Reply:
non-blocking means there is no timeout, it will always immediately return.